Adds post-process cli guide, specifying vafs by element name (#400)
stfc/janus-core: docs/source/user_guide/command_line.rst

For all options, run ``janus md --help``.

Post-processing
+++++++++++++++

> After MD velocity autocorrelation functions (VAFs) and radial distribution functions (RDFs) may be calculated as a post-processing step. It is possible to build partial VAFs and RDFs from arbitrary sequences of atom indices or species names.

Reviewer: ```suggestion
After MD, velocity autocorrelation functions (VAFs) and radial distribution functions (RDFs) may be calculated as a post-processing step. It is possible to build partial VAFs and RDFs from arbitrary sequences of atom indices or species names.
```

To compute the VAF across all atoms from the command line the following options could be passed:

.. code-block:: bash

   janus md --ensemble nve --struct tests/data/NaCl.cif --steps 1000 --traj-every 10 --post-process-kwargs "{'vaf_compute': True}"

> In this case a file ``NaCl-nve-T300.0-vaf.dat`` would be create containing correlation values and lag times. In this case there will be 100 values with an implied time step of 10 fs.

Reviewer: Worth saying here or above explicitly that the post processing uses the trajectory file (assuming I'm remembering correctly that is does)?

Author: yes and it does look to be the case as in md.py

```python
data = read(self.traj_file, index=":")
# ...
compute_vaf(
                data,
                out_paths,
                use_velocities=use_vel,
                fft=fft,
                index=slice_,
                atoms_filter=self.post_process_kwargs.get("vaf_atoms", None),
            )
```

To compute partial VAFs of Na and Cl the following kwargs may be passed:

.. code-block:: bash

   janus md --ensemble nve --struct tests/data/NaCl.cif --steps 100 --traj-every 10 --post-process-kwargs "{'vaf_compute': True, 'vaf_atoms': (('Na',), ('Cl',)), 'vaf_output_files': ('vaf_na.dat', 'vaf_cl.dat')}"

> Where ``'vaf_atoms'`` is a ``Sequence`` of ``Sequence`` of element names (or atom indices) included in each VAF. The output files  must also be specified in the case of multiple VAFs, these are by default relative to the working directory.

Reviewer: ```suggestion
Where ``'vaf_atoms'`` is a ``Sequence`` of ``Sequence`` of element names (or atom indices) included in each VAF. The output files must also be specified in the case of multiple VAFs, by default defined relative to the working directory.
```

For all options, run ``janus md --help``.

Post-processing
+++++++++++++++

After MD, velocity autocorrelation functions (VAFs) and radial distribution functions (RDFs) may be calculated as a post-processing step. It is possible to build partial VAFs and RDFs from arbitrary sequences of atom indices or species names.

To compute the VAF across all atoms from the command line the following options could be passed:

.. code-block:: bash

   janus md --ensemble nve --struct tests/data/NaCl.cif --steps 1000 --traj-every 10 --post-process-kwargs "{'vaf_compute': True, 'vaf_start': 10, 'vaf_stop': None, 'vaf_step': 2}"

In this case a file ``NaCl-nve-T300.0-vaf.dat`` would be created containing correlation values and lag times. The file will contain 45 correlation values with an implied time step of 20 fs. That is starting at time step 100 and correlating every 20 steps.

> The VAF is computed from the trajectory file, therefore ``--traj-every`` controls the correlation possible temporal resolutions. The precise steps used in the VAF may be controlled by the ``vaf_start``, ``vaf_stop``, and ``vaf_step`` options.

Reviewer: ```suggestion
The VAF is computed from the trajectory file, so ``--traj-every`` controls the possible temporal resolution of the correlations. The precise steps used in the VAF may be controlled by the ``vaf_start``, ``vaf_stop``, and ``vaf_step`` options.
```

> The VAF is computed from the trajectory file, therefore ``--traj-every`` controls the correlation possible temporal resolutions. The precise steps used in the VAF may be controlled by the ``vaf_start``, ``vaf_stop``, and ``vaf_step`` options.

Reviewer: I'm not sure if there's a clearer way of saying this, but it could be read as a contradictory to say that the resolution depends on the trajectory, but also that the "precise steps" can be controlled. Maybe I'd just drop "precise"?

A second question: how does the `vaf_step` interact with the `traj_every`? Is it the step, within the steps saved in the trajectory, or MD steps? If the latter, what happens if `vaf_step` is not a multiple of `traj_every`?

Author: Yes it is clunky. The trajectory saving being $> 1$ implies a sub-sampling of the timestep, by saving a subset of simulation steps as trajectory frames. 

If you then have a $> 1$ ``vaf_step`` then that in turn sub-samples the sub-sampled timestep, by taking a sub-set of trajectory frames as input. 

I don't think it matters if it is not a multiple. 

Perhaps we just state that ``vaf_step`` is the rate of stepping through the saved trajectory and leave it at that.

Reviewer: > Perhaps we just state that `vaf_step` is the rate of stepping through the saved trajectory and leave it at that.

If that's what it's doing, then yeah I think that (still mentioning the start and stop) is enough, and then yeah being a multiple shouldn't matter either.
